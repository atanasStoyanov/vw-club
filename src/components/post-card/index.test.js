import React from 'react';
import renderer from 'react-test-renderer';
import PostCard from '.';
import TestingEnvironmet from '../../test-utils/router';
jest.mock('../post-values', () => 'PostValues');
jest.mock('../button/link-button', () => 'LinkButton')

describe('PostCard Component', () => {
    it('should render post card with title=Test', () => {

        const tree = renderer.create(
            <TestingEnvironmet value={{
                loggedIn: true,
                user: {
                    id: '123'
                }
            }}>
                <PostCard
                    title='Test'
                    carModel='Test car'
                    _id='123' />
            </TestingEnvironmet>).toJSON();

        expect(tree).toMatchSnapshot();
    })
})