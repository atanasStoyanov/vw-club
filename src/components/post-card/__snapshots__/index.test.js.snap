// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard Component should render post card with title=Test 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <img
      alt="car"
      src="passat-icon.svg"
      style={
        Object {
          "height": "300px",
          "width": "400px",
        }
      }
    />
  </div>
  <div
    className="info"
  >
    <h3>
      Test
    </h3>
    <h4
      className="model"
    >
      Test car
    </h4>
    <PostValues />
    <LinkButton
      href="/forum/post/:123"
      title="Details"
    />
  </div>
</div>
`;
